### PR TITLE
Moodle v4.1 fix

### PR DESCRIPTION
**Pull request by hiddengravitas**
and some error handling to make this less awkward to debug in the future

### CODE REVIEW

**Review by n0toose — APPROVED** [review]
Change was successfully tested by myself and the author.

File: syncmymoodle/__main__.py [hunk 1]
```diff
@@ -178,8 +178,13 @@ def __init__(self, config):
...
     def login(self):
         def get_session_key(soup):
-            session_key = soup.find("a", {"data-title": "logout,moodle"})["href"]
-            return re.findall("sesskey=([a-zA-Z0-9]*)", session_key)[0]
+            try:
+                session_key = soup.find("input", {"name": "sesskey"})["value"]
+            except TypeError:
+                logger.critical("Can't retrieve session key")
+                logger.info(soup)
+                exit(1)
```
Comment (septatrix) [inline syncmymoodle/__main__.py:186]: `exit` is intended for interactive shells, in scripts `sys.exit` should be used. Raising an exception would also be fine by me.

**Review by septatrix — CHANGES_REQUESTED** [review]

File: syncmymoodle/__main__.py (hunk 1 above)
Comment (n0toose) [reply syncmymoodle/__main__.py:186]: Oh, true that.

Comment (hiddengravitas) [reply syncmymoodle/__main__.py:186]: Fair enough. Although I just copied this verbatim (exit(1) is used in 5 other places in __main__.py)

Comment (n0toose) [reply syncmymoodle/__main__.py:186]: Note: We actually use `exit()` in several different places.

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L216

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L323

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L1204

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L1191

https://github.com/Romern/syncMyMoodle/blob/9727e0b8f7d94b3586b015ebc377d0f6adaea457/syncmymoodle/__main__.py#L1231

Comment (septatrix) [reply syncmymoodle/__main__.py:186]: Okay, then just leave it. Better to tackle this in a separate issue :D

**Pull request merged**